app/components/primer/select_menu_component.rb

```diff
@@ -0,0 +1,323 @@
+      @loading = fetch_or_fallback([true, false], loading, DEFAULT_LOADING)
+      @blankslate = fetch_or_fallback([true, false], blankslate, DEFAULT_BLANKSLATE)
+      @list_border = fetch_or_fallback(LIST_BORDER_CLASSES.keys, list_border,
+        DEFAULT_LIST_BORDER_CLASS)
+      @list_role = list_role
+      @message = message
+      @kwargs = kwargs
+      @kwargs[:tag] ||= :div
+      @kwargs[:role] ||= "menu" if @kwargs[:tag] == :"details-menu"
+    end
+
+    def render?
+      items.any? || tabs.any? || content.present? || message.present?
+    end
+
+    private
+
+    def loading?
+      @loading
+    end
+
+    def blankslate?
+      @blankslate
+    end
+
+    class Tab < Primer::Slot
+      DEFAULT_SELECTED = false
+
+      attr_reader :selected
+
+      def initialize(selected: DEFAULT_SELECTED, **kwargs)
+        @selected = fetch_or_fallback([true, false], selected, DEFAULT_SELECTED)
+        @kwargs = kwargs
+        @kwargs[:tag] = :button
+        @kwargs[:classes] = class_names(
+          "SelectMenu-tab",
+          kwargs[:classes],
+        )
+        @kwargs[:"aria-selected"] = "true" if @selected
+      end
+
+      def component
+        Primer::BaseComponent.new(**@kwargs)
+      end
+    end
+
+    class Item < Primer::Slot
+      DEFAULT_TAB = 1
+      DEFAULT_SELECTED = false
+
+      attr_reader :icon, :tab, :divider
+
+      def initialize(selected: DEFAULT_SELECTED, icon: nil, tab: DEFAULT_TAB, divider: nil, **kwargs)
+        @selected = fetch_or_fallback([true, false], selected, DEFAULT_SELECTED)
+        @icon = icon
+        @tab = (tab || DEFAULT_TAB).to_i
+        @divider = divider
+        @kwargs = kwargs
+        @kwargs[:tag] ||= :button
+        @kwargs[:role] ||= if @selected || @icon
+          "menuitemcheckbox"
+        else
+          "menuitem"
+        end
+        @kwargs[:classes] = class_names(
+          "SelectMenu-item",
+          kwargs[:classes],
+        )
+        @kwargs[:"aria-checked"] = "true" if @selected
+      end
+
+      def wrapper_component
+        case @kwargs[:tag]
+        when :button
+          Primer::ButtonComponent.new(**@kwargs)
+        when :a
+          Primer::LinkComponent.new(**@kwargs)
+        else
+          Primer::BaseComponent.new(**@kwargs)
+        end
+      end
+
+      # Private: Only used if `icon` is non-nil.
+      def icon_component
+        Primer::OcticonComponent.new(
+          icon: icon,
+          classes: class_names(
+            "SelectMenu-icon SelectMenu-icon--check",
+            @kwargs[:icon_classes],
+          )
+        )
+      end
+
+      # Private: Only used if `divider` is non-nil.
+      def divider_component
+        divider_classes = class_names(
+          "SelectMenu-divider",
+          @kwargs[:divider_classes],
+        )
+        if divider.is_a?(String)
+          Primer::BaseComponent.new(
+            tag: :div,
+            classes: divider_classes
+          )
```

Comment:
Can't we add the content here somehow so we can avoid the `if/else` here https://github.com/primer/view_components/pull/142/files#diff-95c7ff2a6caa40c12ee9c0d64052014219c2b6653001dbbdb9816657cf90e18cR63 ?

Reply:
I tried that originally but the content didn't appear in the html.